controllers/secret.go

```diff
@@ -94,24 +94,14 @@ func (r *BerglasSecretReconciler) updateSecret(ctx context.Context, req ctrl.Req
 		return nil
 	}
 
-	newSecretData, err := r.resolveBerglasSchemas(ctx, bs.Spec.Data)
+	// When we update both a berglasSecret and a pod which volume the berglasSecret,
+	// the pod might volume secret which is not updated yet.
+	// So, we delete secret firstly, and then create new secret.
+	err := r.Delete(ctx, &secret)
 	if err != nil {
-		return err
-	}
-
-	newAnnotationData, err := json.Marshal(bs.Spec.Data)
-	if err != nil {
-		return err
+		return fmt.Errorf("failed to delete secret: %w", err)
```

Comment:
I think this message is confusing because the user of this method want to update their secret, but message is "failed to delete".
How about following:
```suggestion
		return fmt.Errorf("failed to update secret in the step of deleting old secret: %w", err)
```

Reply:
Thank you!!